gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/source/extractor/extract/kafka/KafkaExtractor.java

```diff
@@ -176,7 +176,12 @@ public D readRecordImpl(D reuse) throws DataRecordException, IOException {
 
           D record = decodeKafkaMessage(nextValidMessage);
 
-          this.statsTracker.onDecodeableRecord(this.currentPartitionIdx, readStartTime, decodeStartTime, nextValidMessage.getValueSizeInBytes());
+          long logAppendTime = 0L;
+          if (nextValidMessage.isTimestampLogAppend()) {
+            logAppendTime = nextValidMessage.getTimestamp();
+          }
+          this.statsTracker.onDecodeableRecord(this.currentPartitionIdx, readStartTime, decodeStartTime,
```

Comment:
I personally found ternary operator in this case: `this.statsTracker.onDecodeableRecord(this.currentPartitionIdx, readStartTime, decodeStartTime,  condition ? 0L : nextValidMessage.getTimestamp()). ` more readable. 

googled a little bit, maybe this is a simple-enough case?  https://stackoverflow.com/questions/160218/to-ternary-or-not-to-ternary.  Don't have strong opinion on this though

Reply:
I prefer the ternary operator too. Had refactored it for debugging and forgot to change it back.